### PR TITLE
Drop support for NumPy 1.20

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Per APE 18, as of today (January, 31 2023) we should drop support for NumPy 1.20 (see https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule). This PR executes that drop of support

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
With dropping the support, we should also remove work-arounds that are needed for numpy 1.20. I think there are not so many:
```
find . -name \*.py -exec grep NUMPY_LT_1_21 {} \; -print
    "NUMPY_LT_1_21_1",
NUMPY_LT_1_21_1 = not minversion(np, "1.21.1")
./astropy/utils/compat/numpycompat.py
                # perhaps axis or where was masked (in NUMPY_LT_1_21 this is
./astropy/utils/masked/core.py
from astropy.utils.compat import NUMPY_LT_1_21_1, NUMPY_LT_1_22
        sys.byteorder == "big" and NUMPY_LT_1_21_1, reason="Numpy GitHub Issue 19153"
./astropy/units/tests/test_quantity_array_methods.py
from astropy.utils.compat import NUMPY_LT_1_21_1
    @pytest.mark.skipif(NUMPY_LT_1_21_1, reason="https://stackoverflow.com/q/69571643")
./astropy/units/tests/test_structured.py
```

Since these are all "mine", I thought I might as well fix them...  Will push a commit shortly, and approve assuming tests pass.

File: tox.ini [hunk 1]
```diff
@@ -74,7 +72,7 @@ deps =
     # The oldestdeps factor is intended to be used to install the oldest versions of all
     # dependencies that have a minimum version.
     oldestdeps: numpy==1.20.*
-    oldestdeps: matplotlib==3.2.*
+    oldestdeps: matplotlib==3.1.*
```
Comment (mhvk) [inline tox.ini:75]: Should have looked better: This seems a mistake!

Comment (mhvk) [reply tox.ini]: Assuming it is a mistake, I just removed that change...

Comment (mhvk) [reply tox.ini]: And changed the numpy line above instead.  All should be good now.

File: astropy/units/tests/test_quantity_array_methods.py [hunk 2]
```diff
@@ -125,9 +124,6 @@ def test_swapaxes(self):
         assert q_swapaxes.unit == q.unit
         assert np.all(q_swapaxes.value == q.value.swapaxes(0, 2))
 
-    @pytest.mark.xfail(
-        sys.byteorder == "big" and NUMPY_LT_1_21_1, reason="Numpy GitHub Issue 19153"
```
Comment (pllim) [inline astropy/units/tests/test_quantity_array_methods.py:129]: @mhvk , won't think comparison also hold for 1.21.0?

Comment (mhvk) [reply astropy/units/tests/test_quantity_array_methods.py:129]: Darn, I'm an idiot! I just thought we were dropping support for the oldest numpy version in `numpycompat.py`, but apparently we never had anything particularly bad about numpy 1.20...  Sigh... 

And thanks for catching that. Am happy to push an update with that commit removed...

**Comment by mhvk** [comment]
I removed my incorrect adjustments! Should now all be fine.

**Comment by mhvk** [comment]
Failures in -dev are all unrelated (asdf), so let's get this in. Thanks, @WilliamJamieson, and apologies again for adding edits for the wrong numpy version!

**Pull request merged**